modules/sdk-coin-sui/src/lib/transaction.ts

```diff
@@ -340,7 +338,7 @@ export class Transaction extends BaseTransaction {
       case SuiTransactionType.PayAllSui:
         coins = k.kind.Single.PayAllSui.coins;
         recipients = [k.kind.Single.PayAllSui.recipient];
-        amounts = [k.kind.Single.PayAllSui.amount];
+        amounts = [0]; // PayAllSui deserialization doesn't return the amount
```

Comment:
can we use an empty value to avoid confusion with a 0-value transfer?

Reply:
updated all!